src/server/saved_objects/migrations/core/migration_context.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * The MigrationOpts interface defines the minimum set of data required
+ * in order to properly migrate an index. MigrationContext expands this
+ * with computed values and values from the index being migrated, and is
+ * serves as a central blueprint for what migrations will end up doing.
+ */
+
+import { buildActiveMappings } from './build_active_mappings';
+import { CallCluster, MappingProperties } from './call_cluster';
+import { VersionedTransformer } from './document_migrator';
+import { fetchInfo, FullIndexInfo } from './elastic_index';
+import { LogFn, Logger, MigrationLogger } from './migration_logger';
+
+export interface MigrationOpts {
+  batchSize: number;
+  pollInterval: number;
+  scrollDuration: string;
+  callCluster: CallCluster;
+  index: string;
+  log: LogFn;
+  mappingProperties: MappingProperties;
+  documentMigrator: VersionedTransformer;
+}
+
+export interface Context {
```

Comment:
nit: The name `Context` seems inconsistent with the interface immediately prior to this (`MigrationOpts`) and the descriptive comment on this file. I would have expected `MigrationContext`